### PR TITLE
feat(containers): add config and better error handling

### PR DESCRIPTION
**Pull request by joshuachp**
Make the container service configurable and implement better logging and
handling of error in the whole runtime.

### CODE REVIEW

File: src/containers.rs [hunk 1]
```diff
@@ -26,23 +26,56 @@ use edgehog_containers::{
...
+    /// Flag to make the container service is required
+    #[serde(default)]
+    required: bool,
+    #[serde(default = "ContainersConfig::default_max_retires")]
+    max_retires: usize,
+}
+
+impl ContainersConfig {
+    const fn default_max_retires() -> usize {
+        10
```
Comment (harlem88) [inline src/containers.rs:48]: Defines `10` as a constant.
```suggestion
        const MAX_RETRIES: i32 = 10;
```

File: src/containers.rs [hunk 2]
```diff
@@ -26,23 +26,56 @@ use edgehog_containers::{
...
 
+/// Configuration for the container service.
+#[derive(Debug, Clone, Copy, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct ContainersConfig {
+    /// Flag to make the container service is required
+    #[serde(default)]
+    required: bool,
+    #[serde(default = "ContainersConfig::default_max_retires")]
+    max_retires: usize,
```
Comment (harlem88) [inline src/containers.rs:43]: typo
```suggestion
    #[serde(default = "ContainersConfig::default_max_retries")]
    max_retries: usize,
```

File: src/containers.rs [hunk 3]
```diff
@@ -26,23 +26,56 @@ use edgehog_containers::{
...
+}
+
+impl ContainersConfig {
+    const fn default_max_retires() -> usize {
```
Comment (harlem88) [inline src/containers.rs:47]:
```suggestion
    const fn default_max_retries() -> usize {
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/containers.rs [hunk 4]
```diff
@@ -26,23 +26,60 @@ use edgehog_containers::{
     store::StateStore,
     Docker,
 };
+use serde::Deserialize;
+use stable_eyre::eyre::eyre;
 use tracing::error;
 
 use crate::controller::actor::Actor;
 
+/// Maximum number of retires for the initialization of the service
```
Comment (harlem88) [inline src/containers.rs:35]:
```suggestion
/// Maximum number of retries for the initialization of the service
```

File: src/containers.rs [hunk 5]
```diff
@@ -58,14 +95,42 @@ where
...
+
+        // retry with an exponential back off
+        while let Err(err) = self.service.init().await {
+            error!(
+                error = format!("{:#}", stable_eyre::Report::new(err)),
+                "couldn't init container service"
+            );
+
+            // Increase the times we tired
+            retires += retires.saturating_add(1);
```
Comment (harlem88) [inline src/containers.rs:109]:
```suggestion
            retries += retries.saturating_add(1);
```

Comment (joshuachp) [reply src/containers.rs:109]: I want to cry

**Review by harlem88 — APPROVED** [review]

**Pull request merged**